smart_home/src/devices/device.cc

```diff
@@ -0,0 +1,9 @@
+#include "../../include/devices/device.h";
+
+namespace devices {
+
+Device::~Device() {
+    delete room_;
```

Comment:
what happens if the room is already delete by someone else? Or in some extreme cases is set to nullptr?